src/net_processing.h

```diff
@@ -232,6 +553,38 @@ class PeerManager final : public CValidationInterface, public NetEventsInterface
 
     int64_t m_stale_tip_check_time; //!< Next time to check for stale tip
 
+    /**
+     * Return the number of outbound peers we
+     * relay transactions to by flooding.
+     * Used to determine whether we should flood to a new peer
+     * which supports reconciliation, in case we haven't reached
+     * the outbound flooding bandwidth-conserving limit.
+     */
+    size_t GetFloodingOutboundsCount() const;
+
+    /**
+     * Transaction reconciliation should happen with peers in the same order,
+     * because the efficiency gain is the highest when reconciliation set difference
+     * is predictable. This queue is used to maintain the order of
+     * peers chosen for reconciliation.
+     */
+    RecursiveMutex cs_recon_queue;
```

Comment:
Again, prefer a non-reenrant mutex and don't use the cs nomenclature.

Reply:
I can switch to non-recursive if UpdateNextReconRequest() takes `m_recon_queue.size()` as an argument, instead of accessing it inside. Do you think that would be preferable?

Reply:
I haven't looked into your branch in great detail, but it looks like `UpdateNextReconRequest()` is only called in one place, which already holds `m_recon_queue_mutex`. Could you make the function `EXCLUSIVE_LOCKS_REQUIRED(m_recon_queue_mutex)` and not retake the mutex inside?